drivers/storage/portworx/component/telemetry.go

```diff
@@ -1059,38 +1061,43 @@ func (t *telemetry) validateTelemetrySSLCert(certBytes []byte, cuuid string) err
 			var ok bool
 			// quick sanity check!
 			if pseudonym, ok = v.Value.(string); !ok {
-				return fmt.Errorf("SSL cert Pseudonym is not a string")
+				return &sec, fmt.Errorf("SSL cert Pseudonym is not a string")
 			}
 			break
 		}
 	}
 	if pseudonym == "" {
 		logrus.Errorf("SSL cert Pseudonym not found")
-		return errInvalidTelemetryCert
+		return &sec, errInvalidTelemetryCert
 	}
 
 	if pseudonym != cuuid {
 		logrus.Errorf("SSL cert Pseudonym %s does not match cluster UUID %s",
 			pseudonym, cuuid)
-		return errInvalidTelemetryCert
+		return &sec, errInvalidTelemetryCert
 	}
-	logrus.Debugf("SSL cert Pseudonym %s matches cluster UUID", pseudonym)
-	return nil
+	logrus.Tracef("SSL cert Pseudonym %s matches cluster UUID", pseudonym)
+	return &sec, nil
 }
 
 // refreshTelemetrySSLCert deletes the telemetry SSL cert secret and telemetry-registration PODs
-func (t *telemetry) refreshTelemetrySSLCert(dep *appsv1.Deployment) {
-	if dep == nil {
+func (t *telemetry) refreshTelemetrySSLCert(sec *v1.Secret) {
+	if sec == nil {
+		return
+	} else if sec.Name != pxutil.TelemetryCertName {
+		logrus.Errorf("invalid secret name %s/%s  (expected %s)", sec.Namespace, sec.Name, pxutil.TelemetryCertName)
 		return
 	}
-	logrus.Warnf("refreshTelemetrySSLCert - deleting telemetry SSL cert secret %s/%s", dep.Namespace, pxutil.TelemetryCertName)
-	err := k8sutil.DeleteSecret(t.k8sClient, pxutil.TelemetryCertName, dep.Namespace, dep.OwnerReferences...)
+
+	logrus.Warnf("refreshTelemetrySSLCert - deleting telemetry SSL cert secret %s/%s", sec.Namespace, sec.Name)
```

Comment:
are we only deleting telemetry cert for refresh? and wit for next reconcilation loop to create it back with owner reference?

Reply:
The secret is created by `px-telemetry-registration` POD, while the owner-reference is supposed to set up by `px-operator`.  So it's not an "atomic create", and things can break and get interrupted.
* also, in normal circumstances, `px-operator` refuses to delete secrets that don't have the `StorageCluster` set as the owner, probably for added security
* some opportunities where things can get interrupted --
  1. by "install PX" quickly followed by "delete PX"  (<< IMO, common when people install PX for the 1st time) -- or
  2. by cycling `spec.telemetry.enabled: true <-> false`

This particular `refreshTelemetrySSLCert` is ran only when we confirmed that the `secret/pure-telemetry-certs` has the wrong ClusterUUID embedded in the SSL certificate -- so we should absolutely NOT use it with Telemetry
* to make things worse, I think Telemetry backend won't try to match the ClusterUUID from the client SSL-cert, with the actual ClusterUUID that's inside the Telemetry data-payload -- and it'll just accept it, and store it against the WRONG CLUSTER!!  (wrong ClusterUUID)
* so, I have no issues calling "unconditional delete" on the `secret/pure-telemetry-certs` -- when we _know_ it contains WRONG DATA

BTW, once we remove the secret, we'll bounce (remove) the `pod/portworx-telemetry-registration` -- so we don't have to wait for the next reconcilation loop as the deployment will restart the POD.